src/core/opamConsole.ml

```diff
@@ -22,6 +22,20 @@ let dumb_term = lazy (
     not Sys.win32
 )
 
+let disable_carriage_return =
+  if (Sys.win32 || Sys.cygwin) && not OpamStd.Sys.tty_out then
+    let old_value_out = OpamStubs.getBinaryMode stdout in
+    let old_value_err = OpamStubs.getBinaryMode stderr in
+    if old_value_err && old_value_out then fun k -> k () else
+    fun k ->
+      set_binary_mode_out stdout true;
+      set_binary_mode_out stderr true;
+      k ();
+      set_binary_mode_out stdout old_value_out;
+      set_binary_mode_out stderr old_value_err
```

Comment:
I think we might want to add a mutex here in case the code is being used in parallel